bigbluebutton-html5/imports/ui/components/audio/audio-modal/component.jsx

```diff
@@ -162,22 +163,30 @@ class AudioModal extends Component {
 
   componentDidMount() {
     const {
-      joinFullAudioImmediately,
-      joinFullAudioEchoTest,
       forceListenOnlyAttendee,
       audioLocked,
+      autoJoin,
+      skipCheck,
+      skipCheckOnJoin,
+      useEchoTest,
     } = this.props;
 
-    if (joinFullAudioImmediately) {
-      this.handleJoinMicrophone();
-    }
+    if (autoJoin) {
+      if (forceListenOnlyAttendee || audioLocked) {
+        return this.handleJoinListenOnly();
+      }
+
+      if (useEchoTest) {
```

Comment:
I like the clarity of this logic, even though you moved it from the container layer to the view layer. If by re-adding the check for isListenOnly enabled this logic becomes cluttered, it would be best to move it back to the container and only pass the conclusions here (like we had before with `joinFullAudioImmediately` `joinFullAudioEchoTest`)